11/Zaubeerer/generators.py

```diff
@@ -0,0 +1,55 @@
+"""
+Turn the following unix pipeline into Python code using generators
+
+$ for i in ../*/*py; do grep ^import $i|sed 's/import //g' ; done | sort | uniq -c | sort -nr
+   4 unittest
+   4 sys
+   3 re
+   3 csv
+   2 tweepy
+   2 random
+   2 os
+   2 json
+   2 itertools
+   1 time
+   1 datetime
+"""
+
+import glob, os
+import re
+
+import collections
+
+def gen_files(pat):
+    """Finds all files"""
+
+    yield from glob.glob(pat)
+
+def gen_lines(files):
+    """Finds all lines"""
+    
+    for file in files:
+        with open(file) as f:
+            yield from f.readlines()
+
+def gen_grep(lines, pattern):
+    """Analyze whether lines match a certain pattern"""
+    for line in lines:
+        if line.startswith(pattern):
+            yield line[len(pattern):-1]
+
+def gen_count(lines):
+    """Count number of occurrences of specific patterns"""
+
+    yield collections.Counter(line for line in lines)
+        
+
+if __name__ == "__main__":
+    # call the generators, passing one to the other
+    file_pattern = '../*/*.py'
```

Comment:
got me files of my homedir, please make configurable

Reply:
See above, could you please clarify?

Reply:
ok yeah that is on us then, we should have better defined.

I mean all good when working in current directory, but not very portable so ideally want to configure the path here. All good for this exercise though.